utils/storage_utils.py

```diff
@@ -0,0 +1,123 @@
+    new_triage_results_dir = os.path.join(base_triage_dir, new_triage_subdir)
+    if cve_model_type == "gru":
+        file_prefix = "gru_model_inference_"
+    elif "bert" in cve_model_type.lower():
+        file_prefix = "bert_model_inference_"
+    else:
+        _logger.info("No Valid model type specified, defaulting to BERT model.")
+        file_prefix = "bert_model_inference_"
+    model_inference_dataset_filename = "{file_prefix}full_output_{new_triage_subdir}_{ecosystem}.csv".format(
+        file_prefix=file_prefix, new_triage_subdir=new_triage_subdir, ecosystem=ecosystem
+    )
+    model_inference_dataset = os.path.join(
+        new_triage_results_dir, model_inference_dataset_filename,
+    )
+    probable_sec_cve_dataset_filename = (
+        "{file_prefix}probable_security_and_cves_"
+        "{new_triage_subdir}_{ecosystem}.csv".format(
+            file_prefix=file_prefix, new_triage_subdir=new_triage_subdir, ecosystem=ecosystem
+        )
+    )
+
+    probable_sec_cve_dataset = os.path.join(
+        new_triage_results_dir, probable_sec_cve_dataset_filename
+    )
+    probable_cve_dataset_filename = "{file_prefix}probable_cves_{new_triage_subdir}_{ecosystem}.csv".format(
+        file_prefix=file_prefix, new_triage_subdir=new_triage_subdir, ecosystem=ecosystem
+    )
+
+    probable_cve_dataset = os.path.join(new_triage_results_dir, probable_cve_dataset_filename,)
+    if not s3_upload:
+        if not os.path.exists(new_triage_results_dir):
+            _logger.info(
+                "Creating New Model Inference Directory: {}".format(new_triage_results_dir)
+            )
+            os.makedirs(new_triage_results_dir)
+        else:
+            _logger.info(
+                "Using Existing Model Inference Directory: {}".format(new_triage_results_dir)
+            )
+
+    if cc.INFERENCE_DROP_DESCRIPTIONS:
+        df.drop(["norm_description", "description"], inplace=True, errors="ignore", axis=1)
+
+    df["triage_is_security"] = 0
+    df["triage_is_cve"] = 0
+    df["triage_feedback_comments"] = ""
+    columns = [
+        "ecosystem",
+        "repo_name",
+        "event_type",
+        "status",
+        "url",
+        "security_model_flag",
+        "cve_model_flag",
+        "triage_is_security",
+        "triage_is_cve",
+        "triage_feedback_comments",
+        "id",
+        "number",
+        "api_url",
+        "created_at",
+        "updated_at",
+        "closed_at",
+        "creator_name",
+        "creator_url",
+    ]
+
+    if not cc.INFERENCE_DROP_DESCRIPTIONS:
+        # This is useful to retain data for retraining.
+        columns += ["description", "norm_description"]
+
+    df = df[columns]
+    _save(
+        df, new_triage_subdir, model_inference_dataset, model_inference_dataset_filename, s3_upload
+    )
+
+    # Now save the probable securities dataset.
+    df = df[df.security_model_flag == 1].drop(["triage_is_cve"], axis=1)
```

Comment:
Before saving dataframe can we update ecosystem in dataframe as currently its taking golang everywhere.